src/languages/lib/kws_swift.js

```diff
@@ -240,7 +240,7 @@ export const identifierHead = either(
   /[\u2C00-\u2DFF\u2E80-\u2FFF]/,
   /[\u3004-\u3007\u3021-\u302F\u3031-\u303F\u3040-\uD7FF]/,
   /[\uF900-\uFD3D\uFD40-\uFDCF\uFDF0-\uFE1F\uFE30-\uFE44]/,
-  /[\uFE47-\uFFFD]/
+  /[\uFE47-\uFEFE\uFF00-\uFFFD]/ // Should be /[\uFE47-\uFFFD]/, but we have to exclude FEFF.
```

Comment:
I assume you are concating this with something else in one of your rules.  You're probably doing something like:

```
\uFEFF+[group that includes \uFEFF*]
```

This is the same problem as `\s*\s*` In a worse case the engine will chase it's tail trying to figure out whether which portion of \uFEFF are matched by the left side rather than the right.  The ambiguity must be removed.

Reply:
The new ranges should be fine as they exclude FEFF, no?

Reply:
Yes, that's one way to fix it if it's not that big a deal.  I was just explaining it to perhaps improve your understanding since you asked about it in the main thread.

Reply:
Fixing poly cases properly can definitely be super annoying. :)  Evidently it's a much larger problem in Prism (they have 100s of cases)... I was able to fix ALL our backtracking issues (~30 or so I think) in a day.